app/controllers/scim_rails/scim_users_controller.rb

```diff
@@ -168,5 +168,13 @@ def flat_keys_to_nested(hash)
         leaf[key_parts.last] = value
       end
     end
+
+    def find_user_by_key(key, user_params)
+      find_by_params = Hash.new
+      find_by_params[key] = user_params[key]
+      find_by_params[ScimRails.config.basic_auth_model.to_s.underscore] = @company
+
+      ScimRails.config.scim_users_model.unscoped.find_or_initialize_by(find_by_params)
```

Comment:
You can simplify it a little as
ScimRails.config.scim_users_model.unscoped.find_or_initialize_by(key: user_params[key], ScimRails.config...: @company)